zeroconf/__init__.py

```diff
@@ -1422,15 +1433,29 @@ def __init__(self, zc: 'Zeroconf') -> None:
         self.name = "zeroconf-Reaper_%s" % (getattr(self, 'native_id', self.ident),)
 
     def run(self) -> None:
+        """Perodic removal of expired entries from the cache."""
         while True:
-            self.zc.wait(10 * 1000)
+            with self.zc.reaper_condition:
+                self.zc.reaper_condition.wait(60)
+
             if self.zc.done:
                 return
-            now = current_time_millis()
-            for record in self.zc.cache.entries():
-                if record.is_expired(now):
-                    self.zc.update_record(now, record)
-                    self.zc.cache.remove(record)
+            try:
+                # We try to iterate the cache without copying the whole
+                # cache as this can be quite an expensive operation.
+                self._cleanup_cache(self.zc.cache.iterable_entries())
+            except RuntimeError:
```

Comment:
Can you also say how often you get `RuntimeError` here in production? This'd be good to know.

Reply:
I haven't actually been able to make it happen but I know its possible.

Reply:
Fair enough, I was worried the retrying might be an issue performance-wise if it happens too often.